lib/src/local_storage.dart

```diff
@@ -113,3 +116,34 @@ class HiveLocalStorage extends LocalStorage {
         .put(supabasePersistSessionKey, persistSessionString);
   }
 }
+
+/// local storage to store pkce flow code verifier.
+class SharedPreferencesGotrueAsyncStorage extends GotrueAsyncStorage {
+  SharedPreferencesGotrueAsyncStorage() {
+    _initialize();
+  }
+
+  static const pkceHiveBoxName = 'supabase.pkce';
+
+  late final SharedPreferences _prefs;
+
+  Future<void> _initialize() async {
```

Comment:
What if _initialize() is not yet finished, so `_prefs` is not set and `getItem()` is called? What about using a completer for `_prefs`?

Reply:
Love it!